Sources/Sentry/SentryTracer.m

```diff
@@ -492,30 +517,41 @@ - (void)finishInternal
     [SentryProfiler stopProfilingSpan:self.rootSpan];
 #endif // SENTRY_TARGET_PROFILING_SUPPORTED
 
-    SentryTransaction *transaction = [self toTransaction];
+    [self addMeasurementsAndChildrenToTransaction:_transaction];
```

Comment:
I refactored this to instantiate the transaction when the tracer is init'd, and just fill in the details of children spans and other measurements here like before. This way we can have the transaction ID if the profiler times out before the transaction goes on to finish and be uploaded.